linkbyondaccount.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+
+define('FROM_MEDIAWIKI', true); //to hook into the phpbbSSO wiki extension
+
+//stuff phpbb wants defined.
+$phpbb_root_path = (defined('PHPBB_ROOT_PATH')) ? PHPBB_ROOT_PATH : './';
+define('IN_PHPBB', true);
+$phpEx = substr(strrchr(__FILE__, '.'), 1);
+
+include_once($phpbb_root_path . 'common.' . $phpEx); //we include the phpbb frame work
+
+$user->session_begin(); //now we let phpbb do all the fancy work of figuring out who the fuck this are.
+$userid = (int)$user->data['user_id'];
+$usertype = $user->data['user_type'];
+
+if ($userid <= 1 || $usertype == 1 || $usertype == 2) {
+    header("location: ucp.php?mode=login&redirect=" . urlencode("linkbyondaccount.php?" . $_SERVER["QUERY_STRING"]));
+    //print_r($user);
+    die();
+}
+
+if (isset($_GET['token']) && strlen($_GET['token']) == 128) {
```

Comment:
Do you know what characters the token may contain? is it just letters/numbers or special characters?

A check that it only contains expected characters would be nice as belt and braces perhaps.